src/types.rs

```diff
@@ -157,10 +166,130 @@ impl std::fmt::Display for InstructionForm {
     }
 }
 
+// Register ---------------------------------------------------------------------------------------
+#[derive(Debug, Clone)]
+pub struct Register {
```

Comment:
Not really related to this PR but since this file gets bigger and bigger might be a good idea to define `Register` and its associated implementations (same for Instruction) in their own dedicated  files.

Reply:
Seems reasonable to me!